File: Servidor/picao/src/main/java/com/example/picao/user/dto/UserResponseDTO.java
package com.example.picao.user.dto;


import com.example.picao.player_profile.dto.PlayerProfileResponseDTO;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UserResponseDTO {
    Integer id;
    String name;
    @JsonProperty("second_name")
    String secondName;
    @JsonProperty("last_name")
    String lastName;
    @JsonProperty("second_last_name")
    String secondLastName;
    @JsonProperty("mobile_number")
    String mobileNumber;
    String email;
    String username;
    @JsonProperty("date_of_birth")
    LocalDate dateOfBirth;
    @JsonProperty("nick_name")
    String nickName;
    @JsonProperty("position_player")
    String positionPlayer;
    @JsonProperty("player_profile")
    PlayerProfileResponseDTO playerProfile;


}

File: Web/golpi/src/app/data/schema/userResponseDto.ts
// Dto
import { GenericDto } from '../../core/models/generic-dto';

export class UserResponseDto extends GenericDto {
  id?: number;
  name?: string;
  second_name?: string;
  last_name?: string;
  second_last_name?: string;
  mobile_number?: string;
  email?: string;
  username?: string;
  date_of_birth?: string | undefined;

  constructor(init?: Partial<UserResponseDto>) {
    super();
    if (init) {
      Object.assign(this, init);
    }
  }
}
